Basic project cleanup tasks

sthewissen opened the pull request.

msftclas commented:
@__sthewissen__,
Thanks for your contribution.
To ensure that the project team has proper rights to use your work, please complete the Contribution License Agreement at https://cla.microsoft.com.

It will cover your contributions to all Microsoft-managed open source projects.
_Thanks,
Microsoft Pull Request Bot_

msftclas commented:
__@sthewissen__, thanks for signing the contribution license agreement. We will now validate the agreement and then the pull request.
<br />_Thanks, Microsoft Pull Request Bot_

jamesmontemagno commented on MVP/MVP/MVP/ViewModels/MyProfileViewModel.cs:
```diff
@@ -75,7 +76,7 @@ public static MyProfileViewModel Instance
         private string _strNotificationsImage = "Notifications.png";
         private string _strSupportImage = "Support.png";
-        private string _strSettingsImage = "Settings.png";
+        private string _strSettingsImage = "icon_settings.png";
```
We need to be careful here. I don't like how these are all setup and we should standardize on a name for all icons.

sthewissen replied:
I'm not even sure why these are being bound. Looks like its done to have the same image be culture specific, but there are better ways to do that I'd say (e.g. a ValueConverter). For cleanliness sake I wouldn't want this kind of stuff in the business part (where it is now) since it's a UI thing.

jamesmontemagno replied:
I added a new document, we should to through ALL of the images, rename them and then do a OnPlatform for UWP for the Assets folder where they actually belong.

jamesmontemagno replied:
I agree, they should NOT be in there at all (the viewmodels that is). It doesn't feel right and isn't necessary.

jamesmontemagno commented on MVP/MVPUI/App.xaml.cs:
```diff
@@ -19,6 +19,8 @@ public partial class App : Application
 
         #region Public Fields
 
+        public static App Instance;
```
App already has an instance to itself with Application.Current

sthewissen replied:
Updated code using Application.Current instead and removed Instance.

jamesmontemagno commented on MVP/MVPUI/LiveIdLogOn.xaml:
```diff
@@ -1,9 +1,13 @@
-﻿<?xml version="1.0" encoding="utf-8" ?>
-<ContentPage xmlns="http://xamarin.com/schemas/2014/forms"
+<?xml version="1.0" encoding="utf-8" ?>
+<local:ModalPage Title="Sign in" xmlns="http://xamarin.com/schemas/2014/forms"
```
Why do we have a modal page? I am going to wrap all modal pages in a navigation page with a close button on the top right.

sthewissen replied:
This was done for iOS because it behaves badly when modal pages and status/navigation bar colors are concerned. What would happen is when you tapped the Sign in button (white status bar) the modal page shows (black status bar) but going back the statusbar won't turn white again. This is fixed through a custom renderer based on ModalPage. If there's a way to check if ContentPage is currently shown modal I'd love to hear it.

jamesmontemagno replied:
So this wont be a problem if we wrap the page in a NavigationPage! I added this to the guidelines. This will ensure that it is correct.

edsnider replied:
Shouldn't "Close"/"Cancel" go in the top left for iOS modals? My understanding is top right is for things like "Save"/"Done" or to take some type of action.

sthewissen replied:
Most likely should. Will change it ASAP.

jamesmontemagno replied:
Yeah, so I think we should not add a custom renderer for that as it isn't really needed and will cause some weird stuff on other platforms.

On the bottom of a page can be Save button and then on the top right will be Close/Cancel always. This just simplifies it across all platforms

jamesmontemagno commented:
I had a few comments, mostly on making sure not to break UWP/Android.

jamesmontemagno commented:
So @sthewissen, we will also need to update UWP and Android icons names to match so they run correctly to. Additionally since all UWP images should be in the "Assets" folder we should update all of them to use the OnPlatform so they display correct.

Are you up to do that in this PR?

sthewissen commented:
Sure, can do that. I still have a whole lot of images in the iOS side of things that I think aren't being used. Let me get rid of those first so I have some sort of baseline to tweak the other ones to :)

jamesmontemagno commented:
I think the contribution type icons that are hard coded should be in a converter.

I appreciate it and we should remove any images that aren't needed from others as it is confusing.

sthewissen commented:
- Moved the type icon to a converter, but it's still working on some string matching magic which I don't consider ideal. For now it's moved as is.
- Cleaned up all unused images for all platforms (tested on Droid and iOS, WinPhone verification needed).
- Removed the activities part on the profile page and moved it to contributions tab

jamesmontemagno commented:
I should be able to test out Windows 10 on Sunday.

jamesmontemagno commented:
Was able to pull this down today.

Only issue was in : SignOutHelper in Android needs a new namespace to be brought in.

This will eventually go away though to be honest as the signout should just bring you back to the sign in page :)

jamesmontemagno commented:
Guess who has permissions to this repo now!!!! <3 I think I figure this stuff out
